### PR TITLE
feat: add preact-snapshot-test.

### LINKED ISSUE #1267: add preact snapshot test
https://github.com/developit/preact/issues/560
https://github.com/developit/preact-render-to-string/issues/16
https://github.com/airbnb/enzyme/pull/742

### PR DESCRIPTION
**Pull request by paranoidjk**
close #1267

- [ ] wait for [jest-serializer-html-string](https://github.com/ruyadorno/jest-serializer-html-string), I will create a PR to fix the problem that html properties not been placed at a newline.

from https://github.com/adriantoine/enzyme-to-json#user-content-serializer:

>This is inspired by jest-serializer-enzyme, I first added a note to jest-serializer-enzyme but I then realised that the output is different, so it is not retro compatible with enzyme-to-json because it's using Enzyme debug helper which doesn't put each prop on a separate line.

First of all, thanks for your contribution! :-)

Please makes sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/1306)

### CODE REVIEW

File: tests/shared/demoTest.js [hunk 1]
```diff
@@ -23,6 +41,14 @@ export function webDemoTest(component, options = {}) {
   });
 }
 
+export function webDemoTest(...args) {
+  if (process.env.TEST_ENV === 'preact') {
```
Comment (cncolder) [inline tests/shared/demoTest.js]: 这里是 'PREACT'

`"test:web-preact": "TEST_ENV=PREACT jest --config .jest.preact.json",`

Comment (paranoidjk) [inline tests/shared/demoTest.js]: ok，大小写倒不限制，主要是手滑两边没统一...

**Comment by afc163** [comment]
snapshot 可能无法覆盖 trigger 和弹层的情况。

**Comment by paranoidjk** [comment]
https://github.com/developit/preact-render-to-string/issues/16#issuecomment-304465081

**Pull request closed without merging**